### PR TITLE
Rename `text_area` to `textarea` and `rich_text_area` to `rich_textarea`

### PR DESCRIPTION
**Pull request by seanpdoyle**
### Motivation / Background

Follow-up to [#52432][]
Related to [#52430][]

Helpers like `text_area_tag` construct [textarea][] elements. This commit renames all view helper and form builder methods to replace occurrences of `text_area` with `textarea`.

### Detail

In the same style, this commit also renames the `rich_text`-prefixed helpers to utilize a `textarea` suffix instead of the existing `text_area` suffix.

To preserve backwards compatibility, this commit defines aliases for the existing `text_area` format.

[#52432]: https://github.com/rails/rails/pull/52432
[#52430]: https://github.com/rails/rails/issues/52430
[textarea]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by byroot** [comment]
This will need a changelog entry. Other than that looks fine to me.

**Comment by seanpdoyle** [comment]
> This will need a changelog entry. Other than that looks fine to me.

There is an `actionview/CHANGELOG.md` change. Are you suggesting one for `actiontext/CHANGELOG.md` as well?

**Comment by seanpdoyle** [comment]
I've pushed up changes to incorporate `actiontext/CHANGELOG.md` entries for the rich text changes, as well as a `railties/CHANGELOG.md` entry to mention changes to generators.

File: actionview/lib/action_view/helpers/form_helper.rb [hunk 1]
```diff
@@ -1682,7 +1683,7 @@ class FormBuilder
       # The methods which wrap a form helper call.
       class_attribute :field_helpers, default: [
         :fields_for, :fields, :label, :text_field, :password_field,
-        :hidden_field, :file_field, :text_area, :checkbox,
+        :hidden_field, :file_field, :textarea, :checkbox,
```
Comment (chaadow) [inline actionview/lib/action_view/helpers/form_helper.rb:1686]: shouldn't we leave both textarea and text_area here?

Comment (byroot) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: I'll take care of it.

Comment (seanpdoyle) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: I followed the precedent established by the `check_box` renaming. 

Do we need entries for both `#check_box` and `#checkbox` as well?

Comment (byroot) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: No, because `:checkbox` is excluded when defining the helpers.

Comment (byroot) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: Done in 3278a40d38d8873f238fe3f66211a6c319f3d97c

Comment (seanpdoyle) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: Thank you!

Comment (chaadow) [reply actionview/lib/action_view/helpers/form_helper.rb:1686]: Thanks!

**Comment by aaronjensen** [comment]
@seanpdoyle @byroot this is reviving a rather old pull request, but this just caught our attention. 

As far as I can tell, this came about in response to the desire to rename `check_box` to `checkbox`. As far as I can tell, "checkbox" is actually a word, and `check_box` was probably never correct. I can understand the rationale for that.

However, when it comes to a text area, while the tag name is `textarea`, the thing that it represents, is a "text area". This is demonstrated (only once) in the MDN docs you linked to: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea#disabled_and_readonly_text_areas

> Disabled and readonly text areas

Unlike "checkbox", "textarea" is not a word. I can only imagine that the reason "textarea" was used in html was that they were avoiding any sort of spaces, hyphens, or necessary capitalization in tag names. 

So the question is, in Rails, are the method names meant to directly tie to the html tag names, or are they more of an English reference. To wit, the method to render a text field is not `input` (which is the actual tag name). I know that's a bit of a stretch, but I think it is pretty clear that the tag name isn't driving the method name.

I think that this is reflected to even more of an extreme when you consider `rich_textarea_tag`, which not only does not emit a `rich-textarea`, but it doesn't even emit a `textarea` afaict. By the standard of naming it after the tag, it should be called `trix_editor_tag` (though we'd have to ignore the `input` hidden tag it renders as well.

Unless I'm missing something, this seems like it may have been a regression in clarity and consistency.

**Pull request merged**